design/01-proc-macro.md

```diff
@@ -0,0 +1,172 @@
+# Procedural Macros
+
+## Summary
+
+We are actively thinking how to allow using user-developed Cairo compiler plugins ([`MacroPlugin`]) in Scarb-powered projects. This post describes our current idea on how to approach this and outlines implementation steps. We would like to invite the community to share feedback!
+
+This document proposes an evolution of Cairo’s compiler plugins system. The primary motivation behind this proposal is to allow dynamic loading of compiler plugins per compilation unit. This document specifies syntax for the definition of procedural macros, a high-level view of their implementation in the compiler, and outlines how they interact with the compilation process.
+
+## Context
+
+### Rust procedural macros
+
+This post assumes reader knowledge on Rust's procedural macros and how they are implemented. A good walkthrough has been published some time ago on IntelliJ Rust blog:
+
+1. https://blog.jetbrains.com/rust/2022/03/18/procedural-macros-under-the-hood-part-i/
+2. https://blog.jetbrains.com/rust/2022/07/07/procedural-macros-under-the-hood-part-ii/
```

Comment:
Looking through rust release notes etc I have not found any information about changes that I would find important. 

I know there is some pre-rfc work on compiling proc-macros to wasm though, see https://internals.rust-lang.org/t/pre-rfc-sandboxed-deterministic-reproducible-efficient-wasm-compilation-of-proc-macros/19359

TBH I feel like even if this articles would in fact become outdated, it would still be just enough of a context for this doc reader. WDYT?

Reply:
If nothing drastic changed, then these articles are relevant. Great!